src/box/memtx_tree.cc

```diff
@@ -684,56 +691,34 @@ tree_iterator_start_raw(struct iterator *iterator, struct tuple **ret)
 		 */
 		memtx_tree_iterator_prev(tree, &it->tree_iterator);
 	}
-
-	if (!equals && (type == ITER_EQ || type == ITER_REQ)) {
-		/*
-		 * Found nothing, iteration will be stopped now. That is the
-		 * last chance to record that the transaction have read the key.
-		 */
-		if (key_is_full) {
-			memtx_tx_track_point(txn, space, idx, it->key_data.key);
-			return 0;
-		}
-		/* it->tree_iterator is positioned on successor of a key! */
-		struct memtx_tree_data<USE_HINT> *res =
-			memtx_tree_iterator_get_elem(tree, &it->tree_iterator);
-		struct tuple *successor = res == NULL ? NULL : res->tuple;
-		memtx_tx_track_gap(txn, space, idx, successor, type,
-				   it->key_data.key, it->key_data.part_count);
-		return 0;
-	}
-
-	struct memtx_tree_data<USE_HINT> *res =
-		memtx_tree_iterator_get_elem(tree, &it->tree_iterator);
-	uint32_t mk_index = 0;
-	if (res != NULL) {
-		*ret = res->tuple;
+	res = memtx_tree_iterator_get_elem(tree, &it->tree_iterator);
+	bool empty_result = !equals && (type == ITER_EQ || type == ITER_REQ);
```

Comment:
I don't like the variable name. If `res == NULL` (or if `clarify` returns NULL) the result of select will be empty even if `empty_result == false`. Also we usually use `is_`, `has_` etc prefix for boolean variable names.
What about `is_empty_eq`?. I'm not sure. I would also be glad if there was a clear comment here.

Reply:
Fixed: changed the variable name to `eq_match` and supplied it with a comment.